.github/workflows/build.yml

```diff
@@ -129,13 +134,80 @@ jobs:
         echo "optflags=-O3 -fno-inline" >> $GITHUB_ENV
       if: matrix.name == 'debug'
 
+    - name: Install Clang 18 (debug asan)
+      run: |
+        set -ex;
+        wget -qO- https://apt.llvm.org/llvm-snapshot.gpg.key | sudo tee /etc/apt/trusted.gpg.d/apt.llvm.org.asc
+        DISTRO_CODENAME="$(lsb_release --codename --short)"
+        sudo add-apt-repository -y "deb http://apt.llvm.org/$DISTRO_CODENAME/ llvm-toolchain-$DISTRO_CODENAME-18 main"
+        sudo apt update -y
+        sudo apt install -y clang-18
+      if: matrix.name == 'debug' && startsWith(matrix.os, 'ubuntu')
+    - name: Set global configure flags (debug asan)
+      run: |
+        # These flags get applied to both the Ruby compilation _and_ the compilation of dependencies
+        # wiht ASAN enabled
+        echo "optflags=-O3 -fno-inline -fno-omit-frame-pointer" >> $GITHUB_ENV
+        # Compile with ASAN
+        echo "debugflags=-fsanitize=address -ggdb3 -fPIC" >> $GITHUB_ENV
+        # Requires clang 18
+        echo "CC=clang-18" >> $GITHUB_ENV
+      if: matrix.name == 'debug' && startsWith(matrix.os, 'ubuntu')
+    - name: Build ASAN dependencies (debug asan)
```

Comment:
This part is quite long and making the worklow hard to read, can you extract it to a `.sh` script?

Reply:
I can, but actually the workflow doesn't even check out this repo! So I'll have to add a step to do that.